File: HW2/HW2codes/HW2ForStudents.m
clc;
clear all;
close all;

% input for the example in the lecture notes
E = 70e3; % E-modulus
BC = [1 2 5 6; 0 0 0 0]; %boundary conditions; form: [degrees of freedom; applied displacements]
F = [0 0 0 -1000 0 0]; % force vector
Con = [1 2; 3 2];  %connectivity matrix; form: [element 1; element 2;etc.]
NodePos = [0 0; 125 100; 0 100]; % node positions; form: [node 1; node 2; etc]
Area = [10 10]; % vector with the area for each element

% functions here
[u, strain, stress, reactions] = HW2functionsExample(E,BC,F,Con,NodePos,Area);

% output: do not change!
for el= 1:size(Con,1)
	fprintf(['the stress in element ', num2str((el)) ' equals ', num2str(stress(el)), ' MPa \n'])
end

for node=1:size(NodePos,1)
	fprintf(['The displacement of node ', num2str(node) , ' in x-direction is ', num2str(u(2*(node)-1)), ' mm \n'])
	fprintf(['The displacement of node ', num2str(node) , ' in y-direction is ', num2str(u(2*(node))), ' mm \n'])
	fprintf(['The reaction force at node ', num2str(node) , ' in x-direction is ', num2str(reactions(2*(node)-1)), ' N \n'])
	fprintf(['The reaction force at node ', num2str(node) , ' in y-direction is ', num2str(reactions(2*(node))), ' N \n']) 
end

File: HW2/HW2codes/HW2ForStudents.py
import HW2functionsExample as HW2f
import numpy as np

# input
E = 70e3 # E-modulus
BC = np.array([[0, 1, 5, 6], [0, 0, 0, 0]]) #boundary conditions; form: [degrees of freedom], [applied displacements]
F = np.array([0, 0, 0, -1000, 0, 0]) # force vector
Con = np.array([[1,2], [3,2]]) # connectivity matrix; form: [[element 1], [element 2],etc]
NodePos = np.array([[0, 0], [125, 100], [0, 100]]) # node position; form: [[element 1],[element 2],etc]
Area = np.array([10, 10]) # vector with the area for each element

# functions here
u, strain, stress, reactions = HW2f.calcUSSR(E,BC,F,Con,NodePos,Area)

# output: do not change!
for el in range(len(Con)):
	print ("the stress in element ", (el+1) , " equals ", "%.2f" % stress[el], "MPa")

for node in range(len(NodePos)):
	print("The displacement of node", node+1 , " in x-direction is ", "%.3f" % u[2*(node+1)-2], "mm")
	print("The displacement of node", node+1 , " in y-direction is ", "%.3f" % u[2*(node+1)-1], "mm")
	print("The reaction force at node", node+1 , " in x-direction is ", "%.0f" % reactions[2*(node+1)-2], "N")
	print("The reaction force at node", node+1 , " in y-direction is ", "%.0f" % reactions[2*(node+1)-1], "N")
	
